### PR TITLE
DOC: #25723 passing kwargs to excel document engine

### LINKED ISSUE #25723: Excel Document Passing Kwargs to Engine
#### Code Sample, a copy-pastable example if possible

```python
xls = pandas.ExcelFile(path, on_demand = True)
sheets = xls.sheet_names
```

#### Problem description

This code used to work, since extra keyword arguments were passed to `xlrd.open_workbook`. As `pd.ExcelFile` is now decoupled from `xlrd` this behaviour was removed. It now returns an error:

```
TypeError: __init__() got an unexpected keyword argument 'on_demand'
```

In a way this is related to #25523, but this is a more general issue, and might or might not solve the problem in the other issue.

#### Expected Output

No error, and the keyword arguments respected. I understand that when other engines are implemented the `on_demand` argument might not make sense, but they might take similar arguments not explicitly made available in the interface of `pandas.ExcelFile`.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.7.2.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 42 Stepping 7, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.24.1
pytest: 4.2.0
pip: 19.0.1
setuptools: 40.7.3
Cython: 0.29.5
numpy: 1.15.4
scipy: 1.2.0
pyarrow: None
xarray: None
IPython: 7.2.0
sphinx: 1.8.4
patsy: 0.5.1
dateutil: 2.7.5
pytz: 2018.9
blosc: None
bottleneck: 1.2.1
tables: 3.4.4
numexpr: 2.6.9
feather: None
matplotlib: 3.0.2
openpyxl: 2.5.14
xlrd: 1.2.0
xlwt: 1.3.0
xlsxwriter: 1.1.2
lxml.etree: 4.3.0
bs4: 4.7.1
html5lib: 1.0.1
sqlalchemy: 1.2.17
pymysql: None
psycopg2: 2.7.6.1 (dt dec pq3 ext lo64)
jinja2: 2.10
s3fs: None
fastparquet: None
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by Dunni3**
- [x] closes #25723
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

This pr tries to address #25723.

Made a small addition to the ExcelFile section of user_guide/io.rst regarding the discussion in #25723.

### CODE REVIEW

File: doc/source/user_guide/io.rst [hunk 1]
```diff
@@ -2864,6 +2864,16 @@ of sheet names can simply be passed to ``read_excel`` with no loss in performanc
     data = pd.read_excel('path_to_file.xls', ['Sheet1', 'Sheet2'],
                          index_col=None, na_values=['NA'])
 
+If control is needed over how a file is read, an ``xlrd`` workbook 
+created with the desired keyword arguments can be passed to ``ExcelFile``.
```
Comment (datapythonista) [inline doc/source/user_guide/io.rst]: May be it's just me, but I don't find this sentence very easy to read.

I'd rephrase to something like: "`pandas.ExcelFile` can also be called with an `xlrd.open_workbook` as a parameter. This lets the user have control on how the workbook is opened, for example with `on_demand=True`". Or something similar.

CC: @WillAyd @TomAugspurger

Comment (TomAugspurger) [reply doc/source/user_guide/io.rst]: @datapythonista's sounds good, though I might change "an xldd.open_workbook" to whatever type of object that returns.

Comment (Dunni3) [reply doc/source/user_guide/io.rst]: I agree. I just committed the suggested changes.

**Review by datapythonista — APPROVED** [review]
lgtm

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Very nice - thanks @Dunni3 !
